src/VisualStudio/Core/Impl/Options/FullSolutionAnalysisOptionBinding.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Options;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Options
+{
+    internal class FullSolutionAnalysisOptionBinding
```

Comment:
it will be only called from UI thread, but object itself doesnt rely on being called from FG. so I dont think we need that assert.